var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -112,10 +182,10 @@ def filter_compilers(self):
             cxx_compiler = self.compiler.cxx
 
         kwargs = {"backup": False, "string": True}
-        nrnmech_makefile = join_path(self.prefix, "./bin/nrnmech_makefile")
+        nrnmech_makefile = join_path(self.prefix, "bin/nrnmech_makefile")
```

Comment:
I think both file filters here could be merged, too.

Reply:
do you meant something like:

```console
-        if spec.satisfies("@:7"):
-            assign_operator = "?="
-        else:
-            assign_operator = "="
-
-        filter_file(
-            "CC {0} {1}".format(assign_operator, env["CC"]),
-            "CC = {0}".format(cc_compiler),
-            nrnmech_makefile,
-            **kwargs,
-        )
-
-        filter_file(
-            "CXX {0} {1}".format(assign_operator, env["CXX"]),
-            "CXX = {0}".format(cxx_compiler),
-            nrnmech_makefile,
-            **kwargs,
-        )
+        assign_operator = "?=" if spec.satisfies("@:7") else "="
+
+        # older neuron versions also use CC
+        compilers = [("CC", "cc_compiler"), ("CXX", "cxx_compiler")]
+
+        for compiler_var, compiler_env in compilers:
+            pattern = "{0} {1} {2}".format(
+                compiler_var, assign_operator, env[compiler_env]
+            )
+            replacement = "{0} = {1}".format(
+                compiler_var, locals()[compiler_env]
+            )
+            filter_file(pattern, replacement, nrnmech_makefile, **kwargs)
```

I am also trying to not add too many changes/improvements to avoid breaking something (this makefile replacement on cray vs non-cray platforms was buggy)

Reply:
Yes. as an added bonus, in the new deployment, no lines match `env[compiler_env]`, and I just matched `^CXX {assign_operator} .*` to get things working "better". Still broken, though. Upstream `neuron` will be pained by the same.

Reply:
Done